src/script.ts

```diff
@@ -29,12 +30,17 @@ export async function downloadAndRunScript(platform: string, url: string, args?:
  * @param platform Operating system of the runner (e.g. "win32" or "linux").
  * @param scriptPath Path to the script (on runner's filesystem).
  */
-export function generateExecCommand(platform: string, scriptPath: string): string {
+export async function generateExecCommand(platform: string, scriptPath: string): Promise<string> {
     // Run the install script using bash
     let installCmd = `bash ${scriptPath}`;
 
     if (platform !== "win32") {
-        installCmd = `sudo -E ${installCmd}`;
+        try {
+            await io.which("sudo", true);
+            installCmd = `sudo -E ${installCmd}`;
+        } catch {
+            // Sudo not available, do not prepend
+        }
```

Comment:
Try to avoid using try/catch blocks for flow control.

Would this work?

```js
sudo = await io.which("sudo", false);
if (sudo) {
    installCmd = `sudo -E ${installCmd}`;
}
```

Reply:
Unfortunately it won't, `await` throws on rejected promises. The two primary ways to handle rejected promises are either to wrap `await` in a `try`-`catch` block or to call the `catch` method on the promise. If we chose the second option we would have to use `then` instead of `await`, and I personally think that makes the code less readable. 

This isn't really for control flow, `io.which` made the design decision to throw if the executable is not on the path when the optional argument is `true`. I don't much like the decision but it does seem to be the intended use.

Reply:
Ah, I missed that you switched the optional argument to `false` in your code snippet. In that case it would work, but I'm a little bit uncomfortable going against what seems to be the intended API of the "@actions/io" library. It probably doesn't matter though in this case

Reply:
Looks like they designed the API to match the `which` behavior here: https://linux.die.net/man/1/which, including erroring if the value is not found (when the second arg is `true`) so it probably isn't problematic to ignore that